js/mixins/dom_iterable.ts

```diff
@@ -0,0 +1,55 @@
+import { DomIterable } from "../dom_types";
+
+// tslint:disable:no-any
+type Constructor<T = {}> = new (...args: any[]) => T;
+
+/** Mixes in a DOM iterable methods into a base class, assumes that there is
+ * a private data iterable that is part of the base class, located at
+ * `[dataSymbol]`.
+ */
+export function DomIterableMixin<K, V, TBase extends Constructor>(
+  // tslint:disable-next-line:variable-name
+  Base: TBase,
+  dataSymbol: symbol
+): TBase & Constructor<DomIterable<K, V>> {
+  // we have to cast `this` as `any` because there is no way to describe the
+  // Base class in a way where the Symbol `dataSymbol` is defined.  So the
+  // runtime code works, but we do lose a little bit of type safety.
+
+  return class DomIterable extends Base {
```

Comment:
Might be nit-picky, but probably when I try printing out created class it would always look like `DomIterable { ... }` instead of something like `Headers { ... }` ?

Reply:
So this is the challenge with this pattern of mixin classes in TypeScript.  It is complicated to change the `.name` on the constructor function that results in this.  I can try to copy it from the `Base`.  I haven't tried in a long time, but let's see.

Reply:
This works now.